docs/docs/uploads.md

```diff
@@ -0,0 +1,693 @@
+
+```ts title="api/src/services/profiles/profiles.ts"
+// highlight-next-line
+import { uploadsProcessors } from 'src/lib/uploads'
+
+export const updateProfile: MutationResolvers['updateProfile'] = async ({
+  id,
+  input,
+}) => {
+  // highlight-next-line
+  const processedInput = await uploadsProcessors.processProfileUploads(input)
+
+  // input.avatar (File) becomes a string 👇
+  // The configuration on where it was saved is passed when we setup uploads in src/lib/uploads.ts
+  // processedInput.avatar -> '/mySavePath/profile/avatar/generatedId.jpg'
+
+  return db.profile.update({
+    data: processedInput,
+    where: { id },
+  })
+}
+```
+
+For each of the models you configure when you setup uploads (in `UploadConfig`) - you have processors for them.
+
+So if you passed:
+
+```ts
+const uploadConfig: UploadsConfig = {
+  profile: {
+    fields: ['avatar'],
+  },
+  anotherModel: {
+    fields: ['signedDocument'],
+  },
+}
+
+const { uploadProcessors } = setupUploads(uploadConfig)
+
+// Available methods 👇
+uploadProcessors.processProfileUploads(profileGqlInput)
+uploadProcessors.processAnotherModelUploads(anotherModelGqlInput)
+
+uploadProcessors.processFileList(arrayOfFiles)
+```
+
+:::info
+You might have already noticed that the processors sort-of tie your GraphQL inputs to your Prisma model.
+
+In essence, these utility functions expect to take an object very similar to the Prisma data argument (the data you're passing to your `create`, `update`), but with File objects at fields `avatar`, and `signedDocument` instead of strings. 
+
+If your `File` is in a different key (or a key did you did not configure in the upload config), it will be ignored and left as is.
+:::
+
+### Processing File lists
+
+If you would like to upload FileLists (or an arrays of Files), use this special processor to save your Files to path strings. This is necessary because String arrays aren't supported on databases - you probably want to save them to a different table, or specific fields.
+
+Let's say you define in your SDL, a way to send an Array of files.
+
+```graphql
+input UpdateAlbumInput {
+  name: String
+  photos: [File]
+}
+```
+
+You can use the file list processor like this:
+
+```ts title="api/src/services/albums.ts"
+export const updateAlbum = async ({
+  id,
+  input,
+}) => {
+
+  // notice we're passing in the file list, and not the input!
+  // highlight-next-line
+  const processedInput = await uploadProcessors.processFileList(input.photos)
+  /* Returns an array like this:
+  [
+  '/baseStoragePath/AG1258019MAFGK.jpg',
+  '/baseStoragePath/BG1059149NAKKE.jpg',
+  ]
+  */
+
+  const mappedPhotos = processedInput.map((path) => ({ path }))
```

Comment:
I like how you included an example of what `processedInput` could look like, so I continued that pattern here

```suggestion
  const mappedPhotos = processedInput.map((path) => ({ path }))
  // Will make `mappedPhotos` be an array of objects like this:
  // [
  //   { path: '/baseStoragePath/AG1258019MAFGK.jpg' },
  //   { path: '/baseStoragePath/BG1059149NAKKE.jpg' },
  // ]
```